Dataset.py

```diff
@@ -242,13 +245,19 @@ def sliding_window(self, xr):
   # noinspection PyMethodMayBeStatic
   def preprocess(self, seq):
     """
+    TODO: This seems to be only called in a function of CachedDataset, which
+      is then only called in HDFDataset. Seems to be completely useless,
+      as there is no override at any point.
+
     :type seq: numpy.ndarray
     :rtype: numpy.ndarray
     """
     return seq
 
   def is_cached(self, start, end):
     """
+    TODO: Is only called within the Dataset itself, can be protected or even private
```

Comment:
Well, I see potential usages also from outside for this. Although this might not be correctly implemented by all datasets. So maybe prefix with `_` (the convention for protected in Python; there is nothing for private).